Sources/SwiftDocCUtilities/Action/Actions/InitAction.swift

```diff
@@ -0,0 +1,156 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SwiftDocC
+
+/// An action that generates an article-only documentation catalog from a template seed.
+public struct InitAction: Action {
+    
+    enum Error: DescribedError {
+        case catalogAlreadyExists, cannotCreate
+        var errorDescription: String {
+            switch self {
+                case .catalogAlreadyExists: return "A documentation catalog with the same name already exists in the output directory."
+                case .cannotCreate: return "Unable to generate the documentation catalog."
+            }
+        }
+    }
+    
+    private let catalogOutputPath: String
+    private let defaultDocumentationTitle: String
+    private let documentationTitle: String
+    private let catalogOutputPathPlaceholder: String
+    private var templateLibraryPath: String
+    private var templateTutorialBaseFolderPath: String
+    private var templateCatalogBaseFolderPath: String
+    private let includeTutorial: Bool
+    private let diagnosticEngine: DiagnosticEngine
+    
+    private let logHandle: LogHandle
+    // TODO: Refactor to properly get the DocC root
+    let templatePath = (
+        Bundle.main.bundleURL
```

Comment:
If we can avoid adding resources then we don't have to update the toolchain submission script to copy/install those resources.

Reply:
@d-ronnqvist Is there any problem with making updates to the script? If we're unable to include additional resources I'll have to reconsider how to implement the docc init command, as the current implementation relies on the availability  of templates as resources

Reply:
I think updating the build script _might_ be more trouble than its worth. It will make testing this feature much more difficult. Did you consider taking the approach SwiftPM does and storing templates as strings?

https://github.com/apple/swift-package-manager/blob/main/Sources/Workspace/InitPackage.swift#L560

Reply:
I made a new implementation adding the template as strings I the source code 😄, resolved here https://github.com/apple/swift-docc/pull/719/commits/cf17375335243bf03d728f2ef33f5b95894493ba